src/lara-app/components/runtime.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from "react";
+import * as firebase from "firebase/app"
+import "firebase/firestore"
+import { Experiment } from "../../shared/components/experiment";
+import { Uploader } from "../../mobile-app/components/uploader";
+import { FirebaseJWT } from "./app";
+import { IExperiment, IExperimentData } from "../../shared/experiment-types";
+
+const QRCode = require("qrcode-svg")
+
+import css from "./runtime.module.scss";
+
+interface Props {
+  experiment: IExperiment;
+  runKey: string;
+  firebaseJWT: FirebaseJWT;
+  setError: (error: any) => void;
+}
+
+export const RuntimeComponent = ({experiment, runKey, firebaseJWT, setError} : Props) => {
+  const [experimentData, setExperimentData] = useState<IExperimentData|undefined>();
+  const [queriedFirestore, setQueriedFirestore] = useState(false);
+  const [qrCode, setQRCode] = useState("")
+  const [manualEntry, setManualEntry] = useState(false)
+
+  useEffect(() => {
+    firebase
+      .firestore()
+      .collection(`runs/${runKey}/experiments`)
+      .orderBy("createdAt", "desc")
+      .limit(1)
+      .onSnapshot(snapshot => {
+        setQueriedFirestore(true);
+        setExperimentData(snapshot.docs[0]?.data?.().data as IExperimentData | undefined);
+
+        // generate QR code
+        const runData = Base64.encode(JSON.stringify(firebaseJWT.claims))
```

Comment:
The claims are typed in the code:

```
// NOTE: this is only a partial description of the returned data, containing only the fields we are interested in
export interface IFirebaseJWT {
  claims: {
    platform_id: string;      // "https://app.rigse.docker",
    platform_user_id: number; // 9,
    user_id: string;          // "https://app.rigse.docker/users/9",
    class_hash: string;       // "31f6344410d9e5874e085df0afa048604ee0131912112c7a",
    offering_id: number       // 13
  };
}
```

I'm passing them as is so that if we update the Lara code to add claims they will be automatically added to the run info.